src/datepicker/ngb-date.ts

```diff
@@ -0,0 +1,25 @@
+export class NgbDate {
+  static from(date: {year: number, month: number, date?: number}) {
+    return date ? new NgbDate(date.year, date.month, date.date ? date.date : 1) : null;
+  }
+
+  constructor(public year: number, public month: number, public date: number) {}
+
+  equals(other: NgbDate) {
+    return other && this.year === other.year && this.month === other.month && this.date === other.date;
+  }
+
+  before(other: NgbDate) {
+    return !other ? false : this.year === other.year ?
+                    this.month === other.month ? this.date === other.date ? false : this.date < other.date :
+                                                 this.month < other.month :
+                    this.year < other.year;
+  }
+
+  after(other: NgbDate) {
+    return !other ? false : this.year === other.year ?
+                    this.month === other.month ? this.date === other.date ? false : this.date > other.date :
+                                                 this.month > other.month :
+                    this.year > other.year;
+  }
+}
```

Comment:
It's shorter and I have tests :) Auto-formatting is kinda crappy, otherwise it's readable. Will note the `compare` and see how `if`s look